### ISSUE TITLE
Error in mode-latex getType

### ISSUE THREAD

**Issue by kurtslagle** [ajaxorg/ace#5090]
When using `mode-latex`, a console error appears saying `Cannot read properties of null (reading 'type')`, coming from https://github.com/ajaxorg/ace-builds/blob/master/src-noconflict/mode-latex.js#L142

It seems like whenever there isn't any content after the cursor, typing `\begin` or `\end` to begin or end a new environment causes a console error to show up.

It's simple to reproduce in the kitchen sink: https://ajaxorg.github.io/ace-builds/kitchen-sink.html
Just set the mode to `LaTeX` and with this document content

```latex
\usepackage{amsmath}
\title{\LaTeX}
\date{}
<start typing here>
```

begin typing `\begin{document}`. Once you finish typing `begin`, the console error will show up.

This doesn't appear to happen if there is any text after where you would be typing `\begin{...`, like this

```latex
<start typing here>
some other stuff
```
typing `\begin` does not make the error appear.

**Comment by andrewnester** [comment]
Released in 1.16.0

**Comment by kurtslagle** [comment]
Thank you for fixing that so quickly!
